pachyderm/values.yaml

```diff
@@ -1,116 +1,256 @@
+      # 'RequestErrors', 'EventStreamBody', or 'all'
+      # (case-insensitive).  See 'AWS SDK for Go' docs for details.
+      # logOptions is analogous to the --obj-log-options argument to
+      # pachctl deploy.
       logOptions: ""
-      noVerifySSL: false
-      iamRole: "" #TODO - ensure iam role wired up in all places
+      # maxUploadParts sets the maximum number of upload parts.  It is
+      # analogous to the --max-upload-parts argument to pachctl
+      # deploy.
+      maxUploadParts: 10000
+      # verifySSL performs SSL certificate verification.  It is the
+      # inverse of the --no-verify-ssl argument to pachctl deploy.
+      verifySSL: true
+      # partSize sets the part size for object storage uploads.  It is
+      # analogous to the --part-size argument to pachctl deploy.
+      partSize: 5242880
+      # region sets the AWS region to use.
+      region: ""
+      # retries sets the number of retries for object storage
+      # requests.  It is analogous to the --retries argument to
+      # pachctl deploy.
+      retries: 10
+      # reverse reverses object storage paths.  It is analogous to the
+      # --reverse argument to pachctl deploy.
+      reverse: true
+      # secret sets the Amazon secret access key to use.  Together with id
+      # and token, it implements the functionality of the
+      # --credentials argument to pachctl deploy.
+      secret: ""
+      # timeout sets the timeout for object storage requests.  It is
+      # analogous to the --timeout argument to pachctl deploy.
+      timeout: "5m"
+      # token optionally sets the Amazon token to use.  Together with
+      # id and secret, it implements the functionality of the
+      # --credentials argument to pachctl deploy.
+      token: ""
+      # uploadACL sets the upload ACL for object storage uploads.  It
+      # is analogous to the --upload-acl argument to pachctl deploy.
+      uploadACL: "bucket-owner-full-control"
+      vault:
+        address: ""
+        role: ""
+        token: ""
+    # backend configures the storage backend to use.  It must be one
+    # of GOOGLE, AMAZON, MINIO, MICROSOFT or LOCAL.
+    backend: ""
+    google:
+      bucket: ""
+      # cred is a GCP service account private key, in object (JSON or
+      # YAML) form, e.g.:
+      #
+      # {
+      #  "type": "service_account",
+      #  "project_id": "…",
+      #  "private_key_id": "…",
+      #  "private_key": "-----BEGIN PRIVATE KEY-----\n…\n-----END PRIVATE KEY-----\n",
+      #  "client_email": "…@….iam.gserviceaccount.com",
+      #  "client_id": "…",
+      #  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+      #  "token_uri": "https://oauth2.googleapis.com/token",
+      #  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+      #  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/…%40….iam.gserviceaccount.com"
+      # }
+      cred:
+        type: "service_account"
```

Comment:
It should accept this cred as one long string instead of being broken out (for developer ease of use)

Reply:
Here are the ergonomics I'm thinking of

```
helm template blah --set-file storage.google.cred=creds.json
```

Reply:
Done. I was able to cheat a little bit, by using YAML’s multi-line strings. Let me know what you think — it may be a bit too clever.

Reply:
Too clever. While I didn't know you could use yaml in the google creds, it frequently gets parsed into a single line string, which is much easier to grok if it's json and not yaml. I think the yaml here is likely confusing. Can you add a note about passing in the file via `helm template blah --set-file storage.google.cred=creds.json` to the comment?

Reply:
I switched it over to JSON and documented it.